src/Payum/Core/Registry/AbstractRegistry.php

```diff
@@ -52,10 +51,17 @@ public function getStorage($class)
         $class = is_object($class) ? get_class($class) : $class;
 
         // TODO: this is a quick fix. I have to find a better\clean solution.
-        if (class_exists($class) && interface_exists(Proxy::class)) {
-            $rc = new \ReflectionClass($class);
-            if ($rc->implementsInterface(Proxy::class)) {
-                $class = $rc->getParentClass()->getName();
+        if (class_exists($class)) {
+            if (interface_exists('Doctrine\Persistence\Proxy')) {
```

Comment:
Hi @StudioMaX allow me to ask, why not using here the approach: `Proxy::class` instead `'Doctrine\Persistence\Proxy'` as string? I think using `::class` is a more modern approach, don't you?

Thanks in advance for your help on this.

Reply:
As I remember this is because we need to check an existence of both `Doctrine\Persistence\Proxy` and `Doctrine\Common\Persistence\Proxy`, which means that we need to import one of them via an alias, like:
```php
use Doctrine\Persistence\Proxy;
use Doctrine\Common\Persistence\Proxy as ProxyOld;
...
if (interface_exists(Proxy::class)) {
   ...
elseif (interface_exists(ProxyOld::class)) {
   ...
```

In my opinion this is less readable than a plain old string.

Reply:
Fair enough. Thanks for clarifying it.